### ISSUE TITLE
Deprecation of android-kotlin-extensions - specifically Parcelize.

### ISSUE THREAD

**Issue by jamesarich** [airbnb/epoxy#1079]
android-kotlin-extensions is being deprecated in the latest releases of kotlin (1.4.20).
https://proandroiddev.com/migrating-the-deprecated-kotlin-android-extensions-compiler-plugin-to-viewbinding-d234c691dec7

When attempting to update my app - I ran into an issue building:
```> A failure occurred while executing com.android.build.gradle.internal.tasks.CheckDuplicatesRunnable
   > Duplicate class kotlinx.android.parcel.IgnoredOnParcel found in modules jetified-kotlin-android-extensions-runtime-1.4.10 (org.jetbrains.kotlin:kotlin-android-extensions-runtime:1.4.10) and jetified-kotlin-parcelize-runtime-1.4.20-RC (org.jetbrains.kotlin:kotlin-parcelize-runtime:1.4.20-RC)
     Duplicate class kotlinx.android.parcel.Parceler found in modules jetified-kotlin-android-extensions-runtime-1.4.10 (org.jetbrains.kotlin:kotlin-android-extensions-runtime:1.4.10) and jetified-kotlin-parcelize-runtime-1.4.20-RC (org.jetbrains.kotlin:kotlin-parcelize-runtime:1.4.20-RC)
     Duplicate class kotlinx.android.parcel.Parceler$DefaultImpls found in modules jetified-kotlin-android-extensions-runtime-1.4.10 (org.jetbrains.kotlin:kotlin-android-extensions-runtime:1.4.10) and jetified-kotlin-parcelize-runtime-1.4.20-RC (org.jetbrains.kotlin:kotlin-parcelize-runtime:1.4.20-RC)
     Duplicate class kotlinx.android.parcel.Parcelize found in modules jetified-kotlin-android-extensions-runtime-1.4.10 (org.jetbrains.kotlin:kotlin-android-extensions-runtime:1.4.10) and jetified-kotlin-parcelize-runtime-1.4.20-RC (org.jetbrains.kotlin:kotlin-parcelize-runtime:1.4.20-RC)
     Duplicate class kotlinx.android.parcel.RawValue found in modules jetified-kotlin-android-extensions-runtime-1.4.10 (org.jetbrains.kotlin:kotlin-android-extensions-runtime:1.4.10) and jetified-kotlin-parcelize-runtime-1.4.20-RC (org.jetbrains.kotlin:kotlin-parcelize-runtime:1.4.20-RC)
     Duplicate class kotlinx.android.parcel.TypeParceler found in modules jetified-kotlin-android-extensions-runtime-1.4.10 (org.jetbrains.kotlin:kotlin-android-extensions-runtime:1.4.10) and jetified-kotlin-parcelize-runtime-1.4.20-RC (org.jetbrains.kotlin:kotlin-parcelize-runtime:1.4.20-RC)
     Duplicate class kotlinx.android.parcel.WriteWith found in modules jetified-kotlin-android-extensions-runtime-1.4.10 (org.jetbrains.kotlin:kotlin-android-extensions-runtime:1.4.10) and jetified-kotlin-parcelize-runtime-1.4.20-RC (org.jetbrains.kotlin:kotlin-parcelize-runtime:1.4.20-RC)
```

On doing some digging, the only other usage I can find for Parcelize is within `epoxy` where the kotlin version is locked to 1.4.10, and android-kotlin-extensions is still in use.
specifically here:
https://github.com/airbnb/epoxy/blob/3905f50321b98ad296b4d058b765ebf1fb5f4dea/epoxy-adapter/src/main/java/com/airbnb/epoxy/stickyheader/StickyHeaderLinearLayoutManager.kt#L13

Are there any plans to update kotlin version and remove the dependency on android-kotlin-extensions?

**Comment by elihart** [comment]
We can try to include this fix in the new release. If anyone wants to contribute a PR for it it would expedite the process :D

**Comment by elihart** [comment]
I've included the fix in 4.2.0 - artifacts are being pushed now and should be available shortly.

https://github.com/airbnb/epoxy/pull/1085

**Comment by jomarierafa** [comment]
still facing this issue 
Duplicate class kotlinx.android.parcel.IgnoredOnParcel found in modules jetified-kotlin-android-extensions-runtime-1.4.20 (org.jetbrains.kotlin:kotlin-android-extensions-runtime:1.4.20) and jetified-kotlin-parcelize-runtime-1.4.20-RC (org.jetbrains.kotlin:kotlin-parcelize-runtime:1.4.20-RC)
Duplicate class kotlinx.android.parcel.Parceler found in modules jetified-kotlin-android-extensions-runtime-1.4.20 (org.jetbrains.kotlin:kotlin-android-extensions-runtime:1.4.20) and jetified-kotlin-parcelize-runtime-1.4.20-RC (org.jetbrains.kotlin:kotlin-parcelize-runtime:1.4.20-RC)
Duplicate class kotlinx.android.parcel.Parceler$DefaultImpls found in modules jetified-kotlin-android-extensions-runtime-1.4.20 (org.jetbrains.kotlin:kotlin-android-extensions-runtime:1.4.20) and jetified-kotlin-parcelize-runtime-1.4.20-RC (org.jetbrains.kotlin:kotlin-parcelize-runtime:1.4.20-RC)
Duplicate class kotlinx.android.parcel.Parcelize found in modules jetified-kotlin-android-extensions-runtime-1.4.20 (org.jetbrains.kotlin:kotlin-android-extensions-runtime:1.4.20) and jetified-kotlin-parcelize-runtime-1.4.20-RC (org.jetbrains.kotlin:kotlin-parcelize-runtime:1.4.20-RC)
Duplicate class kotlinx.android.parcel.RawValue found in modules jetified-kotlin-android-extensions-runtime-1.4.20 (org.jetbrains.kotlin:kotlin-android-extensions-runtime:1.4.20) and jetified-kotlin-parcelize-runtime-1.4.20-RC (org.jetbrains.kotlin:kotlin-parcelize-runtime:1.4.20-RC)
Duplicate class kotlinx.android.parcel.TypeParceler found in modules jetified-kotlin-android-extensions-runtime-1.4.20 (org.jetbrains.kotlin:kotlin-android-extensions-runtime:1.4.20) and jetified-kotlin-parcelize-runtime-1.4.20-RC (org.jetbrains.kotlin:kotlin-parcelize-runtime:1.4.20-RC)
Duplicate class kotlinx.android.parcel.WriteWith found in modules jetified-kotlin-android-extensions-runtime-1.4.20 (org.jetbrains.kotlin:kotlin-android-extensions-runtime:1.4.20) and jetified-kotlin-parcelize-runtime-1.4.20-RC (org.jetbrains.kotlin:kotlin-parcelize-runtime:1.4.20-RC)

**Comment by ibcurly** [comment]
same ^

**Comment by choiruru** [comment]
solved with this version
epoxy : 4.1.0
kotlinVersion : 1.4.10

**Comment by hadi-altinawi** [comment]
Adding `kapt "com.airbnb.android:epoxy-processor:$epoxyVersion"` fixed the problem.
Example:

```
def epoxyVersion = '4.3.1'
    implementation "com.airbnb.android:epoxy:$epoxyVersion"
    kapt "com.airbnb.android:epoxy-processor:$epoxyVersion"
```

**Comment by abiramee** [comment]
Worked fine with these
epoxy: 4.1.0
kotlinVersion: 1.4.21

**Comment by jomarierafa** [comment]
i finally solved this by removing android-kotlin-extension on my project

> still facing this issue
> Duplicate class kotlinx.android.parcel.IgnoredOnParcel found in modules jetified-kotlin-android-extensions-runtime-1.4.20 (org.jetbrains.kotlin:kotlin-android-extensions-runtime:1.4.20) and jetified-kotlin-parcelize-runtime-1.4.20-RC (org.jetbrains.kotlin:kotlin-parcelize-runtime:1.4.20-RC)
> Duplicate class kotlinx.android.parcel.Parceler found in modules jetified-kotlin-android-extensions-runtime-1.4.20 (org.jetbrains.kotlin:kotlin-android-extensions-runtime:1.4.20) and jetified-kotlin-parcelize-runtime-1.4.20-RC (org.jetbrains.kotlin:kotlin-parcelize-runtime:1.4.20-RC)
> Duplicate class kotlinx.android.parcel.Parceler$DefaultImpls found in modules jetified-kotlin-android-extensions-runtime-1.4.20 (org.jetbrains.kotlin:kotlin-android-extensions-runtime:1.4.20) and jetified-kotlin-parcelize-runtime-1.4.20-RC (org.jetbrains.kotlin:kotlin-parcelize-runtime:1.4.20-RC)
> Duplicate class kotlinx.android.parcel.Parcelize found in modules jetified-kotlin-android-extensions-runtime-1.4.20 (org.jetbrains.kotlin:kotlin-android-extensions-runtime:1.4.20) and jetified-kotlin-parcelize-runtime-1.4.20-RC (org.jetbrains.kotlin:kotlin-parcelize-runtime:1.4.20-RC)
> Duplicate class kotlinx.android.parcel.RawValue found in modules jetified-kotlin-android-extensions-runtime-1.4.20 (org.jetbrains.kotlin:kotlin-android-extensions-runtime:1.4.20) and jetified-kotlin-parcelize-runtime-1.4.20-RC (org.jetbrains.kotlin:kotlin-parcelize-runtime:1.4.20-RC)
> Duplicate class kotlinx.android.parcel.TypeParceler found in modules jetified-kotlin-android-extensions-runtime-1.4.20 (org.jetbrains.kotlin:kotlin-android-extensions-runtime:1.4.20) and jetified-kotlin-parcelize-runtime-1.4.20-RC (org.jetbrains.kotlin:kotlin-parcelize-runtime:1.4.20-RC)
> Duplicate class kotlinx.android.parcel.WriteWith found in modules jetified-kotlin-android-extensions-runtime-1.4.20 (org.jetbrains.kotlin:kotlin-android-extensions-runtime:1.4.20) and jetified-kotlin-parcelize-runtime-1.4.20-RC (org.jetbrains.kotlin:kotlin-parcelize-runtime:1.4.20-RC)

**Comment by degea9** [comment]
It seems to be fixed on epoxy version 4.4.0. I have built successfully without the error with version 4.4.0.
